lib/taste_tester/commands.rb

```diff
@@ -210,5 +210,179 @@ def self.upload
       logger.error(exception.backtrace.join("\n"))
       exit 1
     end
+
+    def self.impact
+      # Get the repository based on the paths specified in config.rb
+      repo = BetweenMeals::Repo.get(
+        TasteTester::Config.repo_type,
+        TasteTester::Config.repo,
+        logger,
+      )
+      if repo && !repo.exists?
+        fail "Could not open repo from #{TasteTester::Config.repo}"
+      end
+
+      # Compute the set of changes using Between Meals
+      changeset = _find_changeset(repo)
+
+      # Use the changeset computed earlier to find modified roles
+      if TasteTester::Config.use_custom_impact_hook
+        impact_roles = TasteTester::Hooks.custom_impact(changeset)
+      else
+        impact_roles = _find_impact(changeset)
+      end
+
+      if impact_roles.empty?
+        logger.warn('No impacted roles were found.')
+      else
+        logger.warn('The following roles have modified dependencies. ' +
+                    'Please test a host in each of these roles.')
+        impact_roles.each { |r| logger.warn("\t#{r}") }
+      end
+
+      # Do any post processing required on the list of impacted roles
+      unless TasteTester::Config.skip_post_impact_hook
+        TasteTester::Hooks.post_impact(impact_roles)
+      end
+    end
+
+    def self._find_changeset(repo)
+      # We want to compare changes in the current directory (working set)
+      # with the "most recent" commit in the VCS. For SVN, this will be the
+      # latest commit on the checked out repository (i.e. 'trunk'). Git/Hg
+      # may have different tags or labels assigned to the "master" branch,
+      # (i.e. 'master', 'stable', etc.) and should be configured if different
+      # than the defaults.
+      start_ref = ''
+      case repo
+      when BetweenMeals::Repo::Svn
+        start_ref = repo.latest_revision
+      when BetweenMeals::Repo::Git
+        start_ref = TasteTester::Config.vcs_start_ref_git
+      when BetweenMeals::Repo::Hg
+        start_ref = TasteTester::Config.vcs_start_ref_hg
+      end
+
+      end_ref = TasteTester::Config.vcs_end_ref
+
+      changeset = BetweenMeals::Changeset.new(
+        logger,
+        repo,
+        start_ref,
+        end_ref,
+        {
+          :cookbook_dirs =>
+            TasteTester::Config.relative_cookbook_dirs,
+          :role_dir =>
+            TasteTester::Config.relative_role_dir,
+          :databag_dir =>
+            TasteTester::Config.relative_databag_dir,
+        },
+        @track_symlinks,
+      )
+
+      return changeset
+    end
+
+    def self._find_impact(changeset)
```

Comment:
is the whole point of _find_impact just to turn these into a set and then print the list of modified roles and cookbooks? And _find_impact_roles is... actually "expand_dependencies" ?

It seems to me a cleaner API would be something like:

```
changes = _find_changeset
full_changes = expand_dependencies(changes) # make this not print things, just build the data
TasteTester::Hooks.post_process_changes!(full_changes) # does nothing in default case, modifies full_changes if desired
# also probably no reason to have a skip option for ^^
unless TasteTester::Hooks.print_changes(full_changes) # returns `true` if it printed anything and `nil` otherwise
  _print_changes(full_changes)
end
```